lib/axiosHttpClient.ts

```diff
@@ -134,6 +131,22 @@ export class AxiosHttpClient implements HttpClient {
         timeout: httpRequest.timeout,
         proxy: convertToAxiosProxyConfig(httpRequest.proxySettings)
       };
+
+      let axiosClient: AxiosInstance;
+      if (httpRequest.proxySettings) {
+        const agent = tunnel.httpsOverHttp({
```

Comment:
What are the security implications for using HTTPS over HTTP? Is the traffic not encrypted when it travels between the source and the proxy, but then is encrypted between the proxy and the destination?